state/application.go

```diff
@@ -2212,6 +2221,12 @@ func (a *Application) ChangeScale(scaleChange int) (int, error) {
 				return nil, errors.NotValidf("cannot remove more units than currently exist")
 			}
 		}
+
+		update := bson.D{{"$set", bson.D{{"scale", newScale}}}}
+		if len(attachStorage) > 0 {
+			update = bson.D{{"$set", bson.D{{"scale", newScale}, {"unitcount", newScale}}}}
```

Comment:
This don't look right. "unitcount" should not depend on whether we're attaching storage or not. And besides that, for k8s units, unitcount is updated when the pod comes up and calls `UpsertCAASUnit` - it calls `incUnitCountOp()` So setting it here will result in being incremented twice.